### PR TITLE
Quiz Questions Editor

### PR DESCRIPTION
**Pull request by alyssamui**
ADD QUIZ QUESTIONS EDITOR
- implemented questions editor for all 3 question types
- fixed `quizzes.json` mock data to reflect quiz string ids and questions with no ids

Left todo:
 - [ ] Call backend endpoints
 - [ ] Integrate Quiz Editor into overall Quiz Creation / Quiz Details screen 
 - [ ] functionality for publish 

Demo:

https://github.com/mike10911/kanbas-web-dev-warriors/assets/82426237/5aece3bd-b606-43c5-8618-aa2386c7ee42

### CODE REVIEW

File: src/Kanbas/Courses/Quizzes/Editor/QuestionsTab.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  const [published, setPublished] = useState(false);
+  const [notifyUsers, setNotifyUsers] = useState(false);
+  const [questions, setQuestions] = useState<Question[]>([]);
+  const [initialQuestions, setInitialQuestions] = useState<Question[]>([]);
+
+  // TODO: PULL QUESTIONS FROM DB ?
+  useEffect(() => {
+    setQuestions([]);
+    setInitialQuestions([]);
+  }, []);
```
Comment (jpwaves) [inline src/Kanbas/Courses/Quizzes/Editor/QuestionsTab.tsx:19]: is this not already added in the backend that Jess has already pushed?

File: src/Kanbas/Courses/Quizzes/Editor/QuestionsTab.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  const deleteQuestion = (index: number) => {
+    const updatedQuestions = questions.filter((_, idx) => idx !== index);
+    setQuestions(updatedQuestions);
+  };
+
+  const handleSave = () => {
+    // TODO: MAKE BACKEND REQUESTS
+    // post all newQuestions
+    // put all questions
+  };
```
Comment (jpwaves) [inline src/Kanbas/Courses/Quizzes/Editor/QuestionsTab.tsx:48]: same as above

File: src/Kanbas/Courses/Quizzes/Editor/QuestionsTab.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+  const handleCancel = () => {
+    setQuestions(JSON.parse(JSON.stringify(initialQuestions)));
+  };
+
+  const handleSaveAndPublish = () => {
+    handleSave();
+    // TODO: ADD PUBLISH QUIZ
+  };
```
Comment (jpwaves) [inline src/Kanbas/Courses/Quizzes/Editor/QuestionsTab.tsx:57]: same as above

File: src/Kanbas/Courses/Quizzes/index.tsx [hunk 4]
```diff
@@ -1,3 +1,12 @@
-export default function Quizzes() {
-  return <div>test</div>;
-}
+import { Link } from "react-router-dom";
+
+function Quizzes() {
+  return (
+    <div className="flex-fill d-flex gap-5">
+      <Link to={`./1`}>Editor</Link>
```
Comment (jpwaves) [inline src/Kanbas/Courses/Quizzes/index.tsx:7]: should we leave a todo here to remember to update this to the correct quiz id?

**Review by jpwaves — APPROVED** [review]
lgtm, just left a few questions related to integrating your stuff with jess's backend since i'm not sure if you're planning to address that in this pr or a different one

File: src/Kanbas/Courses/Quizzes/Editor/QuestionsTab.tsx (hunk 1 above)
Comment (alyssamui) [reply src/Kanbas/Courses/Quizzes/Editor/QuestionsTab.tsx:19]: oh yeah I was waiting for the cluster to be up to actually implement + test these, I can make a new PR for that

File: src/Kanbas/Courses/Quizzes/index.tsx (hunk 4 above)
Comment (alyssamui) [reply src/Kanbas/Courses/Quizzes/index.tsx:7]: yep I can add that in

File: src/Kanbas/Courses/Quizzes/Editor/QuestionsTab.tsx (hunk 3 above)
Comment (alyssamui) [reply src/Kanbas/Courses/Quizzes/Editor/QuestionsTab.tsx:57]: will address in new PR

**Pull request merged**